File: USACO_Jan2024/BalanceBact.py
from collections import deque
from typing import List, Tuple

# inputing N number of patches of field
N = int(input())

# input each path
a = deque()
for _ in range(N):
    temp = int(input().strip())
    a.append(temp)

# start traversing the queue and tracking the processes in a list of tuples => (power, multiplicity)
processes: List[Tuple[int, int]] = []
ans = 0
for i in range(N):
    # get the first element
    multiplicity = a.popleft()

    # using processes to calculate the real multiplicity using the current multiplicity
    for j in range(len(processes) - 1, -1, -1):
        # update the multiplicity
        multiplicity += ((processes[j][0] - (N-i) + 1) * processes[j][1] * -1)

    # if the multiplicity is 0, then we don't need to do anything
    if multiplicity == 0:
        continue
    # otherwise, we need to add the process to the list
    ans += abs(multiplicity)

    processes.append((N-i, multiplicity))

print(ans)

File: USACO_Jan2024/BalanceBact.java
import java.util.*;

public class BalanceBact {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // inputing N number of patches of field
        int N = scanner.nextInt();

        // input each path
        Queue<Long> a = new LinkedList<>();
        for (int i = 0; i < N; i++) {
            long temp = scanner.nextLong();
            a.add(temp);
        }

        // start traversing the queue and tracking the processes in a List<Pair<Integer, Long>> => <power, multiplicity>
        List<Pair<Integer, Long>> processes = new ArrayList<>();
        long ans = 0;
        for (int i = 0; i < N; i++) {
            // get the first element
            long multiplicity = a.poll();

            // using processes to calculate the real multiplicity using the current multiplicity
            for (int j = processes.size() - 1; j >= 0; j--) {
                // update the multiplicity
                multiplicity += ((processes.get(j).getKey() - (N-i) + 1) * processes.get(j).getValue() * -1);
            }

            // if the multiplicity is 0, then we don't need to do anything
            if (multiplicity == 0) continue;
            // otherwise, we need to add the process to the vector
            ans += Math.abs(multiplicity);

            processes.add(new Pair<>(N-i, multiplicity));
        }

        System.out.println(ans);
        scanner.close();
    }
}

class Pair<K, V> {
    private K key;
    private V value;

    public Pair(K key, V value) {
        this.key = key;
        this.value = value;
    }

    public K getKey() {
        return key;
    }

    public V getValue() {
        return value;
    }
}
